### DISCUSSION TITLE
Add new component "Signal selector"

### DISCUSSION THREAD

**Discussion by Mandrzejy** [FakeFishGames/Barotrauma#14286] [Ideas and suggestions]
I am a simple man. I yearn for nothing more than chaos during my playsessions with friends. I achieve that by building various circuits that would mess with them. My latest project was to create a button that when pressed, would invert the ship's controls. That's when i encountered a problem: there is no component that would allow me to chose exactly one out of two signals provided. The cheapest workaround circuit costs 2 multiply components, 1 not component and 1 adder component. That is to expensive for an average submarine worker.

 Therefore, I humbly ask for a component with 3 inputs and 1 output like so:
input:
which_signal_to_select (1 for signal_1 and 2 for signal_2)
signal_1
signal_2
output:
signal_out (the chosen input)

**Comment by jcx1991911** [comment]
It's already in the unstable build

**Reply by Mandrzejy** [reply]
K, thx for letting me know.

**Comment by gerakis** [comment]
Thats called a multiplexer in real life electronics btw. Also known as a mux.

**Comment by Regalis11** [comment]
This has been added in the unstable version, coming up in the next update. :)
